extensions/arc/src/common/utils.ts

```diff
@@ -107,20 +107,37 @@ export function getDatabaseStateDisplayText(state: string): string {
  * @param name The name of the resource to delete
  * @returns Promise resolving to true if the user confirmed the name, false if the input box was closed for any other reason
  */
-export async function promptForResourceDeletion(namespace: string, name: string): Promise<boolean> {
+
+/**
+ * Opens an input box prompting and validating the user's input.
+ * Used because vscode.window.showInputBox doens't let you set the title.
+ * @param options Options for the input box
+ * @param title An optional title for the input box
+ * @returns Promise resolving to the user's input if it passed validation,
+ * false if the input box was closed for any other reason
+ */
+async function promptInputBox(title: string, options: vscode.InputBoxOptions): Promise<string | false> {
 	const inputBox = vscode.window.createInputBox();
-	inputBox.title = loc.resourceDeletionWarning(namespace, name);
-	inputBox.placeholder = name;
+	inputBox.title = title;
+	inputBox.prompt = options.prompt;
+	inputBox.placeholder = options.placeHolder;
+	inputBox.password = options.password ?? false;
+	inputBox.value = options.value ?? '';
+	inputBox.ignoreFocusOut = options.ignoreFocusOut ?? false;
+
 	return new Promise(resolve => {
 		let valueAccepted = false;
-		inputBox.onDidAccept(() => {
-			if (inputBox.value === name) {
-				valueAccepted = true;
-				inputBox.hide();
-				resolve(true);
-			} else {
-				inputBox.validationMessage = loc.invalidResourceDeletionName(inputBox.value);
+		inputBox.onDidAccept(async () => {
+			if (options.validateInput) {
+				const errorMessage = await options.validateInput(inputBox.value);
+				if (errorMessage) {
+					inputBox.validationMessage = errorMessage;
+					return;
+				}
 			}
+			valueAccepted = true;
+			inputBox.hide();
+			resolve(inputBox.value);
```

Comment:
I don't like mixing return types like this - I'd rather we just resolved to undefined if a value wasn't chosen. (so this part is fine, I'd just say we should update line 144)